scripts/cassandra-integration-test.sh

```diff
@@ -51,7 +51,7 @@ apply_schema() {
 run_integration_test() {
   local version=$1
   local schema_version=$2
-  local cid=$(setup_cassandra ${version})
+  cid=$(setup_cassandra ${version})
```

Comment:
why this change?

Reply:
as in this script at the end we want to remove the cassandra container and we are using trap for that with `EXIT` signal, and inside the trap we are passing `cid` as the argument to the `teardown_cassandra` function. Earlier when we are running this script it was giving below error 
```
scripts/cassandra-integration-test.sh: line 1: cid: unbound variable
```
as the trap is listing for the `EXIT` signal `teardown_cassandra` function only executes once the script completed and by that time we don't have the `cid` variable in the scope as it was Initialized with local keyword which are bound to theat functions only. 
Below is the exam script I wrote to show the trap working with `EXIT` signal
```
sachin@SACHIN-MAURYA:~/tp/script-test $ cat trap-test.sh 
#!/bin/bash

first(){
echo "this is first function"
}

last(){
echo "Last function call"
}

second(){
echo "this is second function"
first
trap 'last' EXIT
}

main(){
echo "starting with main"
second
echo "ending with main"
}

main
```

Output
```
sachin@SACHIN-MAURYA:~/tp/script-test $ bash trap-test.sh 
starting with main
this is second function
this is first function
ending with main
Last function call
```

Reply:
disagree with your analysis, but you're right that there is a bug. The bug is that (a) the trap instruction is at the end of the function and (b) it's using non-interpolated string ''. I think the trap instruction should follow immediately after `local cid=` is assigned, and it should use interpolated string `"teardown_cassandra ${cid}"` which will be resolved at the time when `cid` is in scope.

Reply:
yes, after testing it I found just having `interpolated string` works there is no need to changes the position of the trap call.